Author: Andreas Beckmann <anbe@debian.org>
Description: use KBUILD_CFLAGS and (KBUILD_)LDFLAGS
 allows building a amd64 kernel module with i386 user space
 skip -Werror=* since that breaks how conftest.sh detects stuff

Index: kernel/Kbuild
===================================================================
--- kernel.orig/Kbuild
+++ kernel/Kbuild
@@ -108,6 +108,7 @@ NV_CONFTEST_CMD := /bin/sh $(NV_CONFTEST
 NV_CFLAGS_FROM_CONFTEST := $(shell $(NV_CONFTEST_CMD) build_cflags)
 
 NV_CONFTEST_CFLAGS = $(NV_CFLAGS_FROM_CONFTEST) $(EXTRA_CFLAGS) -fno-pie
+NV_CONFTEST_CFLAGS += $(filter-out -Werror%,$(KBUILD_CFLAGS))
 
 NV_CONFTEST_COMPILE_TEST_HEADERS := $(obj)/conftest/macros.h
 NV_CONFTEST_COMPILE_TEST_HEADERS += $(obj)/conftest/functions.h
Index: kernel/nvidia/nvidia.Kbuild
===================================================================
--- kernel.orig/nvidia/nvidia.Kbuild
+++ kernel/nvidia/nvidia.Kbuild
@@ -92,7 +92,7 @@ always += $(NVIDIA_INTERFACE)
 always-y += $(NVIDIA_INTERFACE)
 
 $(obj)/$(NVIDIA_INTERFACE): $(addprefix $(obj)/,$(NVIDIA_OBJECTS))
-	$(LD) -r -o $@ $^
+	$(LD) $(or $(KBUILD_LDFLAGS),$(LDFLAGS)) -r -o $@ $^
 
 
 #
Index: kernel/Makefile
===================================================================
--- kernel.orig/Makefile
+++ kernel/Makefile
@@ -112,7 +112,7 @@ else
   # cannot be defined in the *Kbuild files, which are only used during stage 1.
 
   %-linux.o: modules
-	$(LD) $(NV_MODULE_COMMON_SCRIPTS) -r -o $@ \
+	$(LD) $(or $(KBUILD_LDFLAGS),$(LDFLAGS)) $(NV_MODULE_COMMON_SCRIPTS) -r -o $@ \
 	  $(subst nv,nvidia,$*).mod.o $(subst nv,nvidia,$*)/$*-interface.o
 
   # Kbuild's "clean" rule won't clean up the conftest headers on its own, and
Index: kernel/nvidia-modeset/nvidia-modeset.Kbuild
===================================================================
--- kernel.orig/nvidia-modeset/nvidia-modeset.Kbuild
+++ kernel/nvidia-modeset/nvidia-modeset.Kbuild
@@ -77,7 +77,7 @@ always += $(NVIDIA_MODESET_INTERFACE)
 always-y += $(NVIDIA_MODESET_INTERFACE)
 
 $(obj)/$(NVIDIA_MODESET_INTERFACE): $(addprefix $(obj)/,$(NVIDIA_MODESET_OBJECTS))
-	$(LD) -r -o $@ $^
+	$(LD) $(or $(KBUILD_LDFLAGS),$(LDFLAGS)) -r -o $@ $^
 
 #
 # Register the conftests needed by nvidia-modeset.ko
